jitsi-media-transform/src/main/kotlin/org/jitsi/nlj/transform/node/CompliancePcapWriter.kt

```diff
@@ -0,0 +1,263 @@
+            return false
+        }
+
+        if (!Files.isDirectory(Paths.get(basePath))) {
+            logger.error("Compliance recording:${capId()} base-path:$basePath is not a valid directory path")
+            return false
+        }
+
+        return true
+    }
+
+    fun writeMetadata(): Boolean {
+        val meta = JSONObject()
+
+        meta.put("endpoint_id", id)
+        meta.put("context_id", contextId)
+        meta.put("capture_end", captureEnd)
+        meta.put("capture_mode", mode)
+
+        val payloadsMap = JSONArray()
+        streamInformationStore.rtpPayloadTypes.forEach {
+            if ((it.value.mediaType == MediaType.AUDIO && (mode == CAP_MODE_AUDIO || mode == CAP_MODE_AUDIO_VIDEO)) ||
+                (it.value.mediaType == MediaType.VIDEO && (mode == CAP_MODE_VIDEO || mode == CAP_MODE_AUDIO_VIDEO))
+            ) {
+                val entry = JSONObject()
+                entry["payload_type"] = it.value.pt
+                entry["encoding"] = it.value.encoding.toString().lowercase()
+                entry["media_type"] = it.value.mediaType.name.lowercase()
+                entry["clock_rate"] = it.value.clockRate
+                payloadsMap.add(entry)
+            }
+        }
+
+        meta.put("payload_map", payloadsMap)
+
+        // json-simple produces one long flat line
+        // File(jsonFilename()).writeText(meta.toJSONString())
+
+        // gson produces elements on separate lines
+        val gson = GsonBuilder().setPrettyPrinting().create()
+        val je: JsonElement = JsonParser.parseString(meta.toJSONString())
+
+        try {
+            File(jsonFilename()).writeText(gson.toJson(je))
+        } catch (ex: Exception) {
+            logger.error("Compliance recording:${capId()} exception: $ex")
+            return false
+        }
+
+        return true
+    }
+
+    fun configure(newMode: String?, newContextId: String?) {
+        // TODO - how to trace long line exceeding 120 chars
+        logger.info("Compliance recording:${capId()} configure mode:$mode -> $newMode")
+        logger.info("Compliance recording:${capId()} contextId:$contextId -> $newContextId")
+
+        // Disable conditions
+        if (newMode.isNullOrEmpty() || newMode == CAP_MODE_NONE) {
+            disable()
+            synchronized(lock) {
+                mode = "none"
+                contextId = null
+            }
+            return
+        }
+
+        // Checks for valid enable conditions
+        if (!validJmtConfig()) {
+            return
+        }
+
+        if (newMode !in listOf(CAP_MODE_AUDIO, CAP_MODE_VIDEO, CAP_MODE_AUDIO_VIDEO)) {
+            logger.error("Compliance recording:${capId()} invalid mode:$newMode")
+            return
+        }
+
+        if (newContextId.isNullOrEmpty()) {
+            logger.error("Compliance recording:${capId()} invalid context-id:$newContextId")
```

Comment:
The logs should already contain CompliancePcapWriter, so no need to include "Complience recording"

Reply:
Done.